### ISSUE TITLE
Failing numpy_test with numpy 1.12 and 1.12.1

### ISSUE THREAD

**Issue by FRidh** [cython/cython#1676]
Python 3.5, Cython 0.25.2 and numpy 1.12.1.
```
======================================================================
FAIL: numpy_test ()
Doctest: numpy_test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/zvm545rqc4d97caqq9h7344bnd06jhzb-python3-3.5.3/lib/python3.5/doctest.py", line 2190, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for numpy_test
  File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/c/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line 1, in numpy_test

----------------------------------------------------------------------
File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/c/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line 196, in numpy_test
Failed example:
    test_point_record()
Expected:
    array([(0.0, 0.0), (1.0, -1.0), (2.0, -2.0)], 
          dtype=[('x', '!f8'), ('y', '!f8')])
Got:
    array([( 0.,  0.), ( 1., -1.), ( 2., -2.)], 
          dtype=[('x', '!f8'), ('y', '!f8')])


======================================================================
FAIL: numpy_test (numpy_test.__test__)
Doctest: numpy_test.__test__.numpy_test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/zvm545rqc4d97caqq9h7344bnd06jhzb-python3-3.5.3/lib/python3.5/doctest.py", line 2190, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for numpy_test.__test__.numpy_test
  File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/c/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line unknown line number, in numpy_test

----------------------------------------------------------------------
File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/c/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line ?, in numpy_test.__test__.numpy_test
Failed example:
    test_point_record()
Expected:
    array([(0.0, 0.0), (1.0, -1.0), (2.0, -2.0)], 
          dtype=[('x', '!f8'), ('y', '!f8')])
Got:
    array([( 0.,  0.), ( 1., -1.), ( 2., -2.)], 
          dtype=[('x', '!f8'), ('y', '!f8')])


======================================================================
FAIL: numpy_test ()
Doctest: numpy_test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/zvm545rqc4d97caqq9h7344bnd06jhzb-python3-3.5.3/lib/python3.5/doctest.py", line 2190, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for numpy_test
  File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/cpp/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line 1, in numpy_test

----------------------------------------------------------------------
File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/cpp/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line 196, in numpy_test
Failed example:
    test_point_record()
Expected:
    array([(0.0, 0.0), (1.0, -1.0), (2.0, -2.0)], 
          dtype=[('x', '!f8'), ('y', '!f8')])
Got:
    array([( 0.,  0.), ( 1., -1.), ( 2., -2.)], 
          dtype=[('x', '!f8'), ('y', '!f8')])


======================================================================
FAIL: numpy_test (numpy_test.__test__)
Doctest: numpy_test.__test__.numpy_test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/nix/store/zvm545rqc4d97caqq9h7344bnd06jhzb-python3-3.5.3/lib/python3.5/doctest.py", line 2190, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for numpy_test.__test__.numpy_test
  File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/cpp/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line unknown line number, in numpy_test

----------------------------------------------------------------------
File "/tmp/nix-build-python3.5-Cython-0.25.2.drv-0/Cython-0.25.2/TEST_TMP/run/cpp/numpy_test/numpy_test.cpython-35m-x86_64-linux-gnu.so", line ?, in numpy_test.__test__.numpy_test
Failed example:
    test_point_record()
Expected:
    array([(0.0, 0.0), (1.0, -1.0), (2.0, -2.0)], 
          dtype=[('x', '!f8'), ('y', '!f8')])
Got:
    array([( 0.,  0.), ( 1., -1.), ( 2., -2.)], 
          dtype=[('x', '!f8'), ('y', '!f8')])


----------------------------------------------------------------------
Ran 10669 tests in 2126.719s
```

**Comment by vbraun** [comment]
Seems to be fixed in dd0b10c7e31ea136632bfcfd46c925adcf4c4af6 (but no release yet)

**Comment by vbraun** [comment]
To clarify: its fixed in dd0b10c7e31ea136632bfcfd46c925adcf4c4af6 and dd9eb10bc5657beed1f66221511c16ec6f4e517d
